### PR TITLE
Added api service

### LINKED ISSUE #155: Create API Service
Create a class, e.g. `Service`, with methods for getting data / sending

### PR DESCRIPTION
**Pull request by PavelKundenko**
## Summary of [issue](https://github.com/ita-social-projects/what-front/issues/155)
Create a class, e.g. Service, with methods for getting data / sending

This PR closes #155 
## Summary of change

- Added API Service class that contains static methods for http method
- Added mentors model

### CODE REVIEW

**Review by archontis-cpu — APPROVED** [review]

File: src/utils/ApiService/ApiService.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    url,
+    method,
+    jwt = '',
+    data = null,
+    headers = {},
+  ) => {
+    const reqHeaders = { ...headers };
+
+    if (jwt) {
+      reqHeaders.Authorization = `Bearer ${jwt}`;
```
Comment (sarhan-azizov) [inline src/utils/ApiService/ApiService.js:16]: could u please use axios interceptors for this ?

File: src/utils/ApiService/ApiService.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  static userAuth = (userData) => ApiService.sendRequest(`${ApiConfig.BASE_URL}${ApiConfig.AUTH_URL}`, 'POST', '', userData);
+
+  static userRegister = (userData) => ApiService.sendRequest(`${ApiConfig.BASE_URL}${ApiConfig.REGISTER_URL}`, 'POST', '', userData);
+
+  static getAccounts = (jwt) => ApiService.sendRequest(`${ApiConfig.BASE_URL}${ApiConfig.ACCOUNTS_URL}`, 'GET', jwt);
+
+  static getUnassignedAccounts = (jwt) => ApiService.sendRequest(`${ApiConfig.BASE_URL}${ApiConfig.NOT_ASSIGNED_URL}`, 'GET', jwt);
+
+  // Courses
+  static addCourse = (jwt, courseData) => ApiService.sendRequest(`${ApiConfig.BASE_URL}${ApiConfig.COURSES_URL}`, 'POST', jwt, courseData);
```
Comment (sarhan-azizov) [inline src/utils/ApiService/ApiService.js:44]: these should be in redux action creators

File: src/utils/ApiService/config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+export const ApiConfig = {
+  BASE_URL: 'http://localhost:3000',
```
Comment (sarhan-azizov) [inline src/utils/ApiService/config.js:2]:
```suggestion
  BASE_URL: 'http://localhost:3000/api',
```

File: src/utils/ApiService/config.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  AUTH_URL: '/api/accounts/auth',
+  REGISTER_URL: '/api/accounts/reg',
+  ACCOUNTS_URL: '/api/accounts',
+  NOT_ASSIGNED_URL: '/api/accounts/NotAssigned',
+  COURSES_URL: '/api/courses',
+  LESSONS_URL: '/api/lessons',
+  LESSON_ASSIGN_URL: '/api/lessons/assign',
+  STUDENT_LESSONS_URL: '/api/lessons/students',
```
Comment (sarhan-azizov) [inline src/utils/ApiService/config.js:10]: I don't see a reason for these constants, do you?

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

**Review by Icemandov — APPROVED** [review]

**Review by Anastasiia0999 — APPROVED** [review]

**Review by so2niko — APPROVED** [review]

**Review by sarhan-azizov — APPROVED** [review]

**Pull request merged**